addons/hr_recruitment/models/hr_recruitment.py

```diff
@@ -78,6 +81,14 @@ def default_get(self, fields):
             self = self.with_context(context)
         return super(RecruitmentStage, self).default_get(fields)
 
+    @api.depends('hired_stage')
+    def _compute_is_warning_visible(self):
+        for stage in self:
+            applicants = self.env['hr.applicant'].search([('stage_id', '=', stage._origin.id)])
```

Comment:
IMO it would be better to do a `read_group()` outside the loop that way only a single query is used to retrieve the number of applicants for all stages :-)